pkg/spoke/addon/lease_controller.go

```diff
@@ -103,12 +108,39 @@ func (c *managedClusterAddOnLeaseController) syncSingle(ctx context.Context,
 	leaseNamespace string,
 	addOn *addonv1alpha1.ManagedClusterAddOn,
 	recorder events.Recorder) error {
+	now := c.clock.Now()
+	gracePeriod := time.Duration(leaseDurationTimes*leaseDurationSeconds) * time.Second
 	// addon lease name should be same with the addon name.
 	observedLease, err := c.leaseLister.Leases(leaseNamespace).Get(addOn.Name)
 
 	var condition metav1.Condition
 	switch {
 	case errors.IsNotFound(err):
+		// for backward compatible, before release-2.3, addons update their leases on hub cluster,
```

Comment:
this will double the lease call with 2.3 hub and 2.2 agent. FYI @TheRealHaoLiu but i think we need this to support version skew. I think we will not need klusterlet-addon-controller to maintain status of managedclusteraddon anymore, right?

@hanqiuzh @itdove @leena-jawale

Reply:
klusterlet-addon-controller also has a progressing=true when installing/create the manifestwork, but for available, we will not touch going forward.